File: lesson_1/src/c_examples/05_arrays.c
#include <stdio.h>

int main() {
    int arr[5] = {1, 2, 3, 4, 5};
    int sum = 0;

    // Compute sum
    for (int i = 0; i < 5; i++) {  // C arrays are 0-indexed
        sum += arr[i];
    }
    printf("Sum of array elements: %d\n", sum);

    // Multiply each element by 2
    for (int i = 0; i < 5; i++) {
        arr[i] *= 2;
    }

    printf("Array after multiplying by 2: ");
    for (int i = 0; i < 5; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    return 0;
}

File: lesson_1/src/fortran_examples/05_arrays.f90
PROGRAM main
INTEGER :: i
INTEGER :: arr(5)
INTEGER :: sum

    ! Initialize array
    arr = (/1, 2, 3, 4, 5/)

    ! Compute sum
    sum = 0
    DO i = 1, SIZE(arr)
        sum = sum + arr(i)
    END DO
    PRINT *, 'Sum of array elements:', sum

    ! Multiply each element by 2
    DO i = 1, SIZE(arr)
        arr(i) = arr(i) * 2
    END DO
    PRINT *, 'Array after multiplying by 2:', arr
END PROGRAM main
